snap/snapcraft.yaml

```diff
@@ -161,32 +161,19 @@ parts:
   postgres-debs:
     plugin: nil
     stage-packages:
-      - perl
-      - perl-base
```

Comment:
Unfortunately, usrmerge and git depends on perl too:

```
snapcraft-charmed-postgresql-on-amd64-for-amd64-2584310 ../project# aptitude why perl-base
i   usrmerge Depends perl-base (>= 5.32.1-3)
...
i   git  Depends perl                         
i A perl Depends perl-base (= 5.34.0-3ubuntu1)
...
i   gpg-agent           Depends init-system-helpers (>= 1.52)
i A init-system-helpers Depends perl-base (>= 5.20.1-3)  
...
i   init                Depends init-system-helpers (>= 1.25)
i A init-system-helpers Depends perl-base (>= 5.20.1-3)
```

So, perl-base cannot be easily removed from SNAP...


BTW, why both core20 and core22 are shipped inside the snap?
```
sudo snapcraft build --shell-after
apt update && apt install ncdu
ncdu /

--- /snap ------
  232.5 MiB [##########] /core22                                                                                                                                                                                                                       
  198.3 MiB [########  ] /core20
  ... 
```

BTW2, is snapcraft is shipped inside the snap?

```
sudo snapcraft build --shell-after

> snapcraft --version
snapcraft 7.3.1.post80+git89fd9ecb           
```

Reply:
> So, perl-base cannot be easily removed from SNAP...
It should still be pulled as a dependency of other packages and should still be available in `/snap/charmed-postgresql/current/usr/bin/`

> BTW, why both core20 and core22 are shipped inside the snap?

I don't think the dev shell is fully isolated. We should be depending only on core22, but my understanding of snap esoterics is limited.